spring.go

```diff
@@ -0,0 +1,405 @@
+package beego
+
+import (
+	"errors"
+	"github.com/astaxie/beego/context"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+const (
+	TagName = "autowired"
+
+	PreHandling  = "preHandling"
+	PostHandling = "postHandling"
+
+	InterfaceType      = "InterfaceType"
+	ImplementationType = "ImplementationType"
+	BeanName           = "BeanName"
+
+	PanicOnError   = "panicOnError"
+	NilOnError     = "nilOnError"
+	SaveErrorToCtx = "saveErrorToCtx"
+)
+
+var (
+	TypeMismatch         = errors.New("type mismatch")
+	FactoryNotFound      = errors.New("factory not found")
+	ErrorHandlerNotFound = errors.New("error handler not found")
+	BeanNameNotSpecified = errors.New("bean name not specified")
+
+	AspectType = aspectType()
+
+	beanNamePattern     = regexp.MustCompile(`beanName\((?P<Value>.+)\)`)
+	bindMethodPattern   = regexp.MustCompile(`bindMethod\((?P<Value>.+)\)`)
+	errorHandlerPattern = regexp.MustCompile(`errorHandler\((?P<Value>.+)\)`)
+	occasionPattern     = regexp.MustCompile(`occasion\((?P<Value>.+)\)`)
+
+	kernel    = make(map[string]Factory)
+	container = make(map[string]reflect.Type)
+
+	panicOnError = new(panicIfError)
+	nilOnError   = new(nilIfError)
+	toCtxOnerror = new(errorToCtx)
+
+	errorHandlers = map[string]ErrorResolver{
+		PanicOnError:   panicOnError,
+		NilOnError:     nilOnError,
+		SaveErrorToCtx: toCtxOnerror,
+	}
+
+	ignoredTypes = []reflect.Type{reflect.TypeOf(new(context.Context))}
+)
+
+type Aspect interface {
+}
+
+type ErrorResolver interface {
+	Handle(*context.Context, error) (breakOperation bool)
+}
+
+type Factory interface {
+	New(*context.Context) (interface{}, error)
+}
+
+func Register(beanName string, factory Factory) {
+	kernel[beanName] = factory
+}
+
+func Resolve(beanName string) Factory {
+	return kernel[beanName]
+}
+
+func RegisterErrorHandler(name string, handler ErrorResolver) {
+	errorHandlers[name] = handler
+}
+
+func RegisterIgnoreType(obj interface{}) {
+	t := reflect.TypeOf(obj)
+	for ; t.Kind() == reflect.Ptr; t = t.Elem() {
+	}
+	ignoredTypes = append(ignoredTypes, t)
+}
+
+func RegisterInterfaceTypeBinding(binding interface{}) {
```

Comment:
warning: exported function RegisterInterfaceTypeBinding should have comment or be unexported (golint)